tcf_website/static/department/sorting.js

```diff
@@ -0,0 +1,216 @@
+    if (!$('#diff-sort-btn').hasClass('active')) {
+        $('#number-sort-btn').removeClass('active')
+        $('#rating-sort-btn').removeClass('active')
+        $('#gpa-sort-btn').removeClass('active')
+        $('#diff-sort-btn').addClass('active')
+    } else {
+        difficulty_A = !difficulty_A;
+    }
+
+    if (difficulty_A) {
+        sort_classes(cmp_by_difficulty_A)
+    } else {
+        sort_classes(cmp_by_difficulty_D)
+    }
+}
+
+const sort_by_gpa = () => {
+    if (!$('#gpa-sort-btn').hasClass('active')) {
+        $('#number-sort-btn').removeClass('active')
+        $('#rating-sort-btn').removeClass('active')
+        $('#diff-sort-btn').removeClass('active')
+        $('#gpa-sort-btn').addClass('active')
+    } else {
+        gpa_A = !gpa_A;
+    }
+
+    if (gpa_A) {
+        sort_classes(cmp_by_gpa_A)
+    } else {
+        sort_classes(cmp_by_gpa_D)
+    }
+}
+
+
+const sort_classes = (cmp) => {
+    $("ul.course-list").each(function (a) {
+        var li = $(this).children("li");
+        li.detach().sort(cmp);
+        $(this).append(li);
+    });
+}
+
+const cmp_by_rating_D = (a, b) => {
+    ratingA = a.querySelectorAll(".mb-0.info")[0].innerHTML.trim();
+    ratingB = b.querySelectorAll(".mb-0.info")[0].innerHTML.trim();
+
+    if (ratingA === '—') {
+        return 1;
+    } else if (ratingB === '—') {
+        return -1;
+    }
+
+    if (ratingA > ratingB) {
+        return -1;
+    } else if (ratingB > ratingA) {
+        return 1;
+    } else {
+        return 0;
+    }
+};
+
+const cmp_by_rating_A = (a, b) => {
+    ratingA = a.querySelectorAll(".mb-0.info")[0].innerHTML.trim();
+    ratingB = b.querySelectorAll(".mb-0.info")[0].innerHTML.trim();
+
+    if (ratingA === '—') {
+        return 1;
+    } else if (ratingB === '—') {
+        return -1;
+    }
+
+    if (ratingA > ratingB) {
+        return 1;
+    } else if (ratingB > ratingA) {
+        return -1;
+    } else {
+        return 0;
+    }
+};
+
+const cmp_by_difficulty_D = (a, b) => {
+    difficultyA = a.querySelectorAll(".mb-0.info")[1].innerHTML.trim();
+    difficultyB = b.querySelectorAll(".mb-0.info")[1].innerHTML.trim();
+
+    if (difficultyA === '—') {
+        return 1;
+    } else if (difficultyB === '—') {
+        return -1;
+    }
+
+    if (difficultyA > difficultyB) {
+        return -1;
+    } else if (difficultyB > difficultyA) {
+        return 1;
+    } else {
+        return 0;
+    }
+};
+
+const cmp_by_difficulty_A = (a, b) => {
+    difficultyA = a.querySelectorAll(".mb-0.info")[1].innerHTML.trim();
+    difficultyB = b.querySelectorAll(".mb-0.info")[1].innerHTML.trim();
+
+    if (difficultyA === '—') {
+        return 1;
+    } else if (difficultyB === '—') {
+        return -1;
+    }
+
+    if (difficultyA > difficultyB) {
+        return 1;
+    } else if (difficultyB > difficultyA) {
+        return -1;
+    } else {
+        return 0;
+    }
+};
+
+const cmp_by_gpa_D = (a, b) => {
```

Comment:
Does this handle when GPA is missing?

Reply:
Not yet, I will update to include an `isNaN` check similar to the others